### PR TITLE
Add overview strategies for EPT hierarchy selection tuning

### LINKED ISSUE #62: Make use of an OverviewStrategy in the DEMRasterSources implementation
At this point `DEMRasterSources` don't use `OverviewStrategy` and rely only on the `PDAL` `EPTReader` capabilities https://github.com/PDAL/PDAL/blob/2.1.0/io/EptReader.cpp#L293-L318. 

This task is to improve control over it to improve real time performance.

The core use case is to be able to select the actual resolution we're querying from the EPT tree.

- Use OverviewStrategy to select the PDAL resolution
- Treat the RasterSource `cellSize` as request into GT `OverviewStrategy` given resolutions computed from metadata
- Use returned resolution as query parameter to PDAL Pipeline

Requires: https://github.com/geotrellis/geotrellis-pointcloud/issues/69

### PR DESCRIPTION
**Pull request by jpolchlo**
The natural mode of operation of the DEM rasterizers is to target the EPT hierarchy level with the closest "cell size" to the requested RasterExtent.  This will aim to satisfy the query with a return density of approximately one point per pixel.  Sadly, this can lead to significant delays in serving queries, particularly when using TIN as the interpolation strategy.

We want to allow for the user to tune queries so that EPT hierarchy levels relative to the raster query can be selected instead.  This would trade off accuracy for performance.

This PR addresses this need by allowing for an OverviewStrategy to be provided to control which level of the EPT tree is selected.  This can be an absolute level, or a level relative to the raster cell size.

Closes #62

### CODE REVIEW

**Comment by jpolchlo** [comment]
This PR should be ready to go.  In principle.  The TIN sources have the overview strategies added in, but they've not been tested, since the procedure for adding this stuff in to geotrellis-pointcloud-server for visual inspection is a pain, and it's a fairly straightforward copy of the strategy code from the IDW sources.  Proposing that we go through the validation procedure for this PR as is.

File: benchmark/src/main/scala/geotrellis/pointcloud/raster/ept/IDWRasterSourceResolutionBench.scala [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    *
+    * jmh:run -i 10 -wi 5 -f1 -t1 .*IDWRasterSourceResolutionBench.*
+    *
+    */
+
+  @Benchmark
+  def IDWRasterSourceNaturalRead(): Option[Raster[MultibandTile]] = rs.read()
+
+  @Benchmark
+  def IDWRasterSourceAuto2Read(): Option[Raster[MultibandTile]] = rsAuto2.read()
```
Comment (pomadchin) [inline benchmark/src/main/scala/geotrellis/pointcloud/raster/ept/IDWRasterSourceResolutionBench.scala]: what is the difference between rs and rsAuto2 here?

**Review by pomadchin — APPROVED** [review]
Looks awesome, thanks! I'd like to merge it after some extra testing done (by me) with GTServer.

File: benchmark/src/main/scala/geotrellis/pointcloud/raster/ept/IDWRasterSourceResolutionBench.scala (hunk 1 above)
Comment (jpolchlo) [reply benchmark/src/main/scala/geotrellis/pointcloud/raster/ept/IDWRasterSourceResolutionBench.scala]: Sorry.  `rsAuto2` was supposed to use the `Auto(2)` overview strategy, but I could never get this benchmark to run without an OOM, so it never got exercised.

Comment (pomadchin) [reply benchmark/src/main/scala/geotrellis/pointcloud/raster/ept/IDWRasterSourceResolutionBench.scala]: @jpolchlo wow, that's concerning, `Auto(2)` is supposed to fetch a less resolute data, do you know why OOM had happened?

Comment (jpolchlo) [reply benchmark/src/main/scala/geotrellis/pointcloud/raster/ept/IDWRasterSourceResolutionBench.scala]: The OOM happened with the default AutoHigherResolution.  Either the VM settings need to be tweaked or the AOI needs to be narrowed to make this test go.  Never got a chance to come back to that work.

Comment (pomadchin) [reply benchmark/src/main/scala/geotrellis/pointcloud/raster/ept/IDWRasterSourceResolutionBench.scala]: @jpolchlo OK, I will take car of it and will launch the benchmark before merging the PR.

Comment (pomadchin) [reply benchmark/src/main/scala/geotrellis/pointcloud/raster/ept/IDWRasterSourceResolutionBench.scala]: I see what is going on, I'll look how we can merge it in or can we at all merge it in with preserving the old behavior / performance.

**Review by pomadchin — APPROVED** [review]

**Comment by pomadchin** [comment]
LGTM, works as expected! Merging and upgrading gt server to use the latest snapshot!

**Pull request merged**